k8s-deploy/pkg/cli/validation.go

```diff
@@ -0,0 +1,319 @@
+package cli
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+)
+
+type any interface{}
+type KeyValue map[string]*TreeNode
+type ListValue []*TreeNode
+
+type TreeNode struct {
+	leaf     bool
+	lineno   int
+	route    []string
+	children KeyValue
+	value    any
+}
+
+type ValidationTree struct {
+	root    *TreeNode
+	yamlMap map[string]interface{}
+	lines   []string
+}
+
+type ValidationManager struct {
+	templateTree, testTree *ValidationTree
+	skippedKeys            []string
+}
+
+func trimComments(t []byte) []byte {
+	// to trim the comments started with "# "
```

Comment:
Sorry, I thought you know that there is a standard way for the comments of each funciton: https://go.dev/doc/effective_go#commentary

Please take this as an example:
```
// Compile parses a regular expression and returns, if successful,
// a Regexp that can be used to match against text.
func Compile(str string) (*Regexp, error) {
```